src/ninja.cc

```diff
@@ -1058,11 +1058,14 @@ int ReadFlags(int* argc, char*** argv,
         options->input_file = optarg;
         break;
       case 'j': {
+        if (!optarg)
+          break;
         char* end;
         int value = strtol(optarg, &end, 10);
-        if (*end != 0 || value <= 0)
+        if (*end != 0 || value < 0)
           Fatal("invalid -j parameter");
-        config->parallelism = value;
+        if (value)
+          config->parallelism = value;
```

Comment:
That just means that `-j` or `-j0` will be cause ninja to use the default parallelism, which is set up in line 1039 above.

I definitely doesn't mean the same as GNU make `-j` or `-j0` i.e. run as many jobs in parallel as are ready to run.

Reply:
True, I could make it INT_MAX. I'll also drop the reference to PR #1140. The reason I mention it is that our build system passes a sanitized version of the `MAKEFLAGS` variable to the command line of the builders. As that by default contains a `-j` in an environment where Make <4.2 has a command called with a `+`-prefixed line, I feel they're related (although not needed to be useful individually).